lib/lcdbwf/tests/testthat/test-results.R

```diff
@@ -0,0 +1,139 @@
+
+# Create objects for testing 'LRT'
+dds_and_res <- make_deseq_results(test='LRT', type=NULL, reduced_design=~1)
+dds <- dds_and_res$dds
+res <- dds_and_res$res$res
+
+lrt_dds_list <- list(dds1=dds)
+lrt_res_list <- list(res1=list(res=res, dds='dds1', label='LRT'))
+lrt_res_list <- lcdbwf:::attach_extra(lrt_res_list, config)
+
+# ------ Test build_results_tabs function ------ #
+test_that("build_results_tabs works with Wald test", {
+  # build_results_tabs requires 'dds_list' in .GlobalEnv
+  dds_list <<- wald_dds_list
+  plots <- suppressWarnings(build_results_tabs(wald_res_list, wald_dds_list, config, text))
+
+  # Check that each plot in the list is a ggplot object
+  for (name in names(plots)) {
+    expect_true(is_ggplot(plots[[name]]$ma_plot))
+    expect_true(is_ggplot(plots[[name]]$volcano_plot))
+    expect_true(is_ggplot(plots[[name]]$pval_hist_plot))
+    # Check diagnostic plots
+    if (config$toggle$results_diagnostics) {
+      # diag_plot_list is a list of ggplot objects
+      for (diag_plot in plots[[name]]$diag_plot_list) {
+        expect_true(is_ggplot(diag_plot))
+      } # for diag_plot
+    } # if config
+  } # for name
+}) # test_that
+
+test_that("build_results_tabs works with LRT", {
+  # build_results_tabs requires 'dds_list' in .GlobalEnv
+  dds_list <<- lrt_dds_list
+  plots <- suppressWarnings(build_results_tabs(lrt_res_list, lrt_dds_list, config, text))
+
+  # Check that each plot in the list is a ggplot object
+  for (name in names(plots)) {
+    expect_true(is_ggplot(plots[[name]]$ma_plot))
+    expect_true(is_ggplot(plots[[name]]$volcano_plot))
+    expect_true(is_ggplot(plots[[name]]$pval_hist_plot))
+    # Check diagnostic plots
+    if (config$toggle$results_diagnostics) {
+      # diag_plot_list is a list of ggplot objects
+      for (diag_plot in plots[[name]]$diag_plot_list) {
+        expect_true(is_ggplot(diag_plot))
+      } # for diag_plot
+    } # if config
+  } # for name
+}) # test_that
+# ---------------------------------------------- #
+
+test_that("check_LRT identifies LRT results correctly", {
+  expect_true(check_LRT(lrt_res_list$res1$res))
+  expect_false(check_LRT(wald_res_list$res1$res))
+})
+
+# Test that mdcat is called with expected values for LRT
+test_that("build_results_tabs calls mdcat with expected character for LRT", {
+  # build_results_tabs requires 'dds_list' in .GlobalEnv
+  dds_list <<- lrt_dds_list
+
+  # Capture mdcat output
+  mdcat_output <<- c()
+  with_mock(
+    `lcdbwf:::mdcat` = mock_mdcat,
+    suppressWarnings(build_results_tabs(lrt_res_list, lrt_dds_list, config, text))
+  )
+
+  expect_true(any(grepl("LRT log2FoldChange values have been set to 0", mdcat_output)))
+})
+
+# Test that mdcat is not called with LRT expected values for Wald test
+test_that("build_results_tabs does not call mdcat with LRT expected character for Wald", {
+  # build_results_tabs requires 'dds_list' in .GlobalEnv
+  dds_list <<- wald_dds_list
```

Comment:
should this be reset at the end of this test so that future tests appended to the bottom don't get `wald_dds_list`? Same for where it's set in the test above (but lrt_dds_list in that case).